### PR TITLE
add a bunch of nothrows

### PR DESCRIPTION
**Pull request by WalterBright**
dmd doesn't throw exceptions, so everything should be `nothrow`. This is some progress with that.

### CODE REVIEW

File: src/dmd/root/array.d [hunk 1]
```diff
@@ -21,7 +21,6 @@ import dmd.root.string;
 // `qsort` is only `nothrow` since 2.081.0
 private extern(C) void qsort(scope void* base, size_t nmemb, size_t size, _compare_fp_t compar) nothrow @nogc;
 
-
```
Comment (thewilsonator) [inline src/dmd/root/array.d:24]: ?

File: src/dmd/common/file.d [hunk 2]
```diff
@@ -52,6 +54,8 @@ struct FileMapping(Datum)
     private const(char)* name;
     // state }
 
+  nothrow:
```
Comment (thewilsonator) [inline src/dmd/common/file.d:57]: I agree with all of them except possibly this one. 

The constructor call `exit(1)` on failure to map. 
This should probably be made to be a call to `fatal()`, which is in the process of being made more "DMD as a library" friendly see https://github.com/dlang/dmd/pull/13891

**Comment by pbackus** [comment]
Same CircleCI error as #13962 and #13963:

> `[2022-04-08T02:47:57.801Z] ['error'] There was an error running the uploader: Error uploading to https://codecov.io: Error: There was an error fetching the storage URL during POST: 404 - Not Found - {'detail': ErrorDetail(string='Build has already finished, uploads rejected.', code='not_found')}`

Looks like the Codecov integration is broken.

File: src/dmd/common/file.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/common/file.d:57]: It's a good point, but it doesn't belong in this PR, which is only about `nothrow`.

**Comment by ibuclaw** [comment]
> Looks like the Codecov integration is broken.

Seems fixed now, rerunning.

**Review by dkorpel — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**